graal/backends/core/analyzers/flake8.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*- the Graal backend.
+#
+# Copyright (C) 2015-2019 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA.
+#
+# Authors:
+#     Nishchith Shetty <inishchith@gmail.com>
+#
+
+import subprocess
+
+from .analyzer import Analyzer
+
+
+class Flake8(Analyzer):
+    """A wrapper for Flake8, a source code style checker for Python."""
+
+    version = '0.2.0'
+
+    def analyze(self, **kwargs):
+        """Add quality checks data using Flake8.
+
+        :param module_path: module path
+        :param details: if True, it returns information about single commit
+
+        :returns result: dict of the results of the analysis
+        """
+        module_path = kwargs['module_path']
+        details = kwargs['details']
+
+        try:
+            msg = subprocess.check_output(['flake8', module_path]).decode("utf-8")
+        except subprocess.CalledProcessError as e:
+            msg = e.output.decode("utf-8")
+        finally:
+            subprocess._cleanup()
+
+        lines = msg.split('\n')
+
+        result = {'warnings': len(lines)}
+
+        if details:
+            result['lines'] = lines
```

Comment:
This part could be improved to process every line separating the file path, the type of warning and the description. 

Another improvement could be to remove from the file path the worktree path. In this case, the best location to do it could be the method `_post`

what do you think?

Reply:
> This part could be improved to process every line separating the file path, the type of warning and the description.

Yeah. That would be better.
```
{
             "file_path": <path>,
              "type_of_warning": E501,
              "line": 124,
              "column": 80,
              "description": "line too long (82 > 79 characters)"
}
```
> Another improvement could be to remove from the file path the worktree path. In this case, the best location to do it could be the method _post
1. So we're not showing the `worktreepath` attribute?
2. I think it'd be better if we handle this in the analyzer as we're already improving the verbosity of each file by iterating and extracting information from each line, only thing the `ModuleAnalyzer`'s `analyze` method will have an extra parameter `worktreepath`.

What do you think of this?

Reply:
> So we're not showing the worktreepath attribute?

probably it's not really useful to know where the tool was executed

> I think it'd be better if we handle this in the ...

sure!